llms/mlx_lm/utils.py

```diff
@@ -113,11 +138,21 @@ def sample(logits: mx.array) -> Tuple[mx.array, float]:
         prob = softmax_logits[0, token]
         return token, prob
 
+    if repetition_penalty < 0 or not isinstance(repetition_penalty, float):
```

Comment:
Nit: I feel the check is a bit too restrictive. Maybe only `repetition_penalty` not being None is enough.

Reply:
`repetition_penalty` is supposed to be greater than 1.0 right? With `repetition_penalty` not being None we will end up accepting values below 1.0, that includes negative float/int

Reply:
I have removed one of the checks in the latest PR.